sysv-pdp11_man/u_man/man1/man.1
.tr ~"
.if t .ds i \(fm\(fm
.if n .ds i ""
.TH MAN 1
.SH NAME
man, manprog \- print entries in this manual
.SH SYNOPSIS
.B man
[ options ] [ section ] titles
.PP
.B /usr/lib/manprog
file
.SH DESCRIPTION
.I Man\^
locates and prints the entry of this manual named
.I title\^
in the specified
.IR section .
(For historical reasons, the word ``page''
is often used as a synonym for ``entry'' in this context.)\ 
The
.I title\^
is entered in lower case.
The
.I section\^
number may not have
a letter suffix.
If no
.I section\^
is specified, the whole manual is searched for
.I title\^
and all occurrences of it are printed.
.I Options\^
and their meanings are:
.PP
.PD 0
.TP 11
.B \-t
Typeset the entry
in the default format (8.5\*i\(mu11\*i).
.TP
.B \-s
Typeset the entry in the small format (6\*i\(mu9\*i).
.TP
.B \-T4014
Display the typeset output on a Tektronix 4014 terminal using
.IR tc (1).
.TP
.B \-Ttek
Same as
.BR \-T4014 .
.TP
.B \-Tst
Print the typeset output on the
.SM MHCC STARE
facility
(this option is not usable on most systems).
.TP
.B \-Tvp
Print the typeset output on a
Versatec printer;
this option is not available at all
.SM UNIX
System sites.
.TP
.BI \-T term\^
Format the entry using
.I nroff
and print it on the standard output
(usually, the terminal);
.I term\^
is the terminal type (see
.IR term (5)
and the explanation below);
for a list of recognized values of
.IR term ,
type
.BR "help term2" .
The default value of
.I term\^
is
.BR 450 .
.TP
.B \-w
Print on the standard output only the
.I "path names\^"
of the entries,
relative to
.BR /usr/man ,
or to the current directory for
.B \-d
option.
.TP
.B \-d
Search the current directory rather than
.BR /usr/man ;
requires the full file name (e.g.,
.BR cu.1c ,
rather than just
.BR cu ).
.TP
.B \-12
Indicates that the manual entry is to be produced in
12-pitch.
May be used when
.SM
.B $TERM
(see below)
is set to one of
.BR 300 ,
.BR 300s ,
.BR 450 ,
and
.BR 1620 .
(The pitch switch on the
.SM DASI
300 and 300s terminals must be manually set to
.B 12
if this option is used.)
.TP
.B \-c
Causes
.I man\^
to invoke
.IR col (1);
note that
.IR col (1)
is invoked automatically by
.I man\^
unless
.I term\^
is one of
.BR 300 ,
.BR 300s ,
.BR 450 ,
.BR 37 ,
.BR 4000a ,
.BR 382 ,
.BR 4014 ,
.BR tek ,
.BR 1620 ,
and
.BR X .
.TP
.B \-y
Causes
.I man\^
to use the non-compacted version of the macros.
.PD
.PP
The above
.I options\^
other than
.BR \-d ,
.BR \-c ,
and
.B \-y
are mutually exclusive, except that the
.B \-s
option may be used in conjunction with the first four
.B \-T
options above.
Any other
.I options\^
are passed to
.IR troff ,
.IR nroff ,
or the
.IR man (5)
macro package.
.PP
When using
.IR nroff ,
.I man\^
examines the environment variable
.SM
.B $TERM
(see
.IR environ (5))
and attempts to select options to
.IR nroff ,
as well as filters,
that adapt the output to the terminal being used.
The
.BI \-T term\^
option overrides the value of
.SM
.BR $TERM\*S ;
in particular, one should use
.B \-Tlp
when sending the output of
.I man\^
to a line printer.
.PP
.I Section\^
may be changed before each
.IR title .
.PP
As an example:
.IP
man \|man
.PP
would reproduce on the terminal this entry,
as well as any other entries named
.I man\^
that may exist in other sections of the manual, e.g.,
.IR man (5).
.PP
If the first line of the input for an entry consists
solely of the string:
.IP
.BI \&\(fm\^\e~\| " x"
.PP
where
.I x\^
is any combination of the three characters
.BR c ,
.BR e ,
and
.BR t ,
and
where there is exactly one blank between the double quote
.RB ( ~ )
and
.IR x ,
then
.I man\^
will preprocess its input through the appropriate combination of
.IR cw (1),
.IR eqn (1)
(\fIneqn\fP for \fInroff\fP)
and
.IR tbl (1),
respectively;
if
.I eqn
or
.I neqn
are invoked, they will automatically read the file
.B /usr/pub/eqnchar
(see
.IR eqnchar (5)).
.PP
The
.I man\^
command executes
.I manprog\^
that takes a file name as its argument.
.I Manprog\^
calculates and returns a string of
three register definitions used by the formatters
identifying the date the file was last modified.
The returned string has the form:
.PP
.RS
.BI \-rd day
.BI \-rm month
.BI \-ry year
.RE
.PP
and is passed to
.I nroff\^
which sets this string as variables for the
.I man\^
macro package.
Months are given from 0 to 11, therefore month is always
1 less than the actual month.
The
.I man\^
macros calculate the correct month.
If the
.I man\^
macro package is invoked as an option to
.IR nroff / troff
(i.e.,
.IR "nroff \-man file" ),
then the current
day/month/year is used as the printed date.
.SH FILES
.PD 0
.TP \w'/usr/man/u_man/man[1-6]/\(**\ \ \ \ 'u
/usr/man/u_man/man[1-6]/\(**
the \fI\s-1UNIX\s+1 System User's Manual\fP
.TP
/usr/man/a_man/man[178]/\(**
the \fI\s-1UNIX\s+1 System Administator's Manual\fP
.TP
/usr/man/local/man[1-8]/\(**
local additions
.TP
/usr/lib/manprog
calculates modification dates of entries
.PD
.SH SEE ALSO
cw(1),
eqn(1),
nroff(1),
tbl(1),
tc(1),
troff(1),
environ(5),
man(5),
term(5).
.SH BUGS
All entries are supposed to be reproducible either on
a typesetter or on a terminal.
However, on a terminal some information is necessarily lost.
.PP
Pages bearing the same name in both manuals will result in the
.I "U\s-1NIX\s+1 System Administrator's Manual"
entry being printed first, if no \fIsection\fP argument is supplied.
.tr ~~
.\"	@(#)man.1	5.2 of 5/18/82


sysvr3/301/usr/src/cmd/man/man.sh
#	Copyright (c) 1984 AT&T
#	  All Rights Reserved

#	THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF AT&T
#	The copyright notice above does not evidence any
#	actual or intended publication of such source code.

#ident	"@(#)man:man.sh	1.4"
#	Catable manual entries printer
#

BASE=/usr

sec=\?
dir=
col=
cmd=
all=
entries=
TTERM=


#  parse options

for i in $*
do case $i in

	[1-8])	sec=$i ;;
	-c)	col=col ;;
	-d)	dir=. ;;
	-w)	cmd=w ;;
	-T*)	TERM="`echo $i | sed 's/-T//'`" ;;
	-12)	TTERM=-12 ;;
	*)	entries="$entries $i" ;;
   esac
done


#  now find the named entries

for i in $entries
do
    if [ "$dir" = "." ]

	then if [ ! -r "$i" ]

		then  echo $0: $i not found >&2
		      exit 1

		else  all="$all $i"

	     fi

	else
	     cd $BASE/catman
	     fil="`find */man$sec/$i.* -print 2>/dev/null`"
	     if [ -n "$fil" ]

		then  all="$all $fil"
		else  echo $0: $i not found >&2
		      exit 1

	     fi
    fi
done


# fix up the terminal type option (make sure to use col, etc.)

case "$TERM$TTERM" in

	300|300s|450|37|300-12|300s-12|450-12| \
		4000a|4000A|382|1620|1620-12)	post="greek -T$TERM$TTERM" ;;
	hp|2621|2645|2640)	col=col; post="greek -T$TERM$TTERM" ;;
	4014|tek)	post="greek -T$TERM$TTERM" ;;
	*)		col=col; post=cat ;;
esac

if [ "$col" != "" ]
   then post="$col | $post"
fi


# now print the pages

if [ "$cmd" = "w" ]

   then	echo $all
	exit 0

   else	for i in $all

	do case $i in

		*.z)	eval "pcat $i | $post" ;;
		*)	eval "cat $i | $post" ;;

	   esac
	done

fi
